splash/browser_tab.py

```diff
@@ -829,11 +834,18 @@ def get(self, url, callback, headers=None, follow_redirects=True):
 
     def _send_request(self, url, callback, method='GET', body=None,
                       headers=None):
-        # XXX: The caller must ensure self._delete_reply is called in a callback.
-        if method != 'GET':
+        # XXX: The caller must ensure self._delete_reply is called in a callback
+        request = self.request_obj(url, headers=headers, body=body)
+        method = method.upper()
+
+        if method not in ["POST", "GET"]:
             raise NotImplementedError()
-        request = self.request_obj(url, headers=headers)
-        reply = self.network_manager.get(request)
+
+        if body and method == "POST":
```

Comment:
hmm I assumed current master code is right when it assumes we can't have POST without body:https://github.com/scrapinghub/splash/blob/master/splash/browser_tab.py#L405 but now I checked this and it seems like this is actually allowed, PyQt does not complain when you pass body=None and it performs POST request okay.

Perhaps in some cases POST without body will be user error (user forgetting to pass body), but given that it is not forbidden by HTTP standard and allowed by PyQt we should probably allow it.